src/map/script.cpp

```diff
@@ -19477,8 +19490,13 @@ BUILDIN_FUNC(unitstopwalk)
 	if (script_hasdata(st, 3))
 		flag = script_getnum(st, 3);
 
-	if(script_rid2bl(2,bl))
+	if(script_rid2bl(2,bl)) {
+		unit_data *ud = unit_bl2ud(bl);
+
+		if (ud != nullptr)
```

Comment:
Should we also add a warning/failure here, if flag != USW_FORCE_STOP and the unit is not walking?